draft-ietf-quic-transport.md

```diff
@@ -1050,8 +1050,14 @@ An endpoint SHOULD ensure that its peer has a sufficient number of available and
 unused connection IDs.  Endpoints store received connection IDs for future use
 and advertise the number of connection IDs they are willing to store with the
 active_connection_id_limit transport parameter.  An endpoint MUST NOT provide
-more connection IDs than the peer's limit.  An endpoint that receives more
-connection IDs than its advertised active_connection_id_limit MUST close the
+more connection IDs than the peer's limit, with the exception that connection
+IDs MAY be provided if sent in a NEW_CONNECTION_ID frame that also requests the
+retirement of enough connection IDs that the number of active connection IDs
+will stay within the limit.  An endpoint that receives a NEW_CONNECTION_ID
+frame that increases the number of active connection IDs - without a Retire
+Prior To field that requests retirement of enough active connections to keep the
+total number of active connection IDs within the value the endpoint advertised
+in its active_connection_id_limit transport parameter - MUST close the
 connection with an error of type CONNECTION_ID_LIMIT_ERROR.
```

Comment:
Maybe this is better:

```suggestion
will stay within the limit.  A NEW_CONNECTION_ID frame might cause an endpoint
might add some active connection IDs and retire others based on the value of the
Retire Prior To field.  After processing a NEW_CONNECTION_ID frame and adding a
retiring active connection IDs, if the number of active connections exceeds the
value advertised in its active_connection_id_limit transport parameter, an
endpoint MUST close the connection with an error of type
CONNECTION_ID_LIMIT_ERROR.
```

Reply:
I can't comment on this change, so writing my comments here. The first sentence needs editorial fixing ("might cause an endpoint might add some active"). But I definitely prefer this version to the original.